File: 2023-2024/sourcecode/mod1a/lab2/62ViewPager2RecyclerViewAdapterJava/app/src/main/java/com/example/a62_viewpager2recyclerviewadapter_java/PagerBasicAdapter.java
package com.example.a62_viewpager2recyclerviewadapter_java;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.a62_viewpager2recyclerviewadapter_java.databinding.PagerItemBinding;

public class PagerBasicAdapter extends RecyclerView.Adapter<PagerBasicViewHolder> {
    @NonNull
    @Override
    public PagerBasicViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new PagerBasicViewHolder(PagerItemBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        ));
    }

    @Override
    public void onBindViewHolder(@NonNull PagerBasicViewHolder holder, int position) {
        holder.bind(DataProvider.values[position]);
    }

    @Override
    public int getItemCount() {
        return DataProvider.values.length;
    }
}

File: 2023-2024/sourcecode/mod1/lab2/61ViewPager2RecyclerViewAdapterKotlin/app/src/main/java/com/example/a61_viewpager2recyclerviewadapter_kotlin/PagerBasicAdapter.kt
package com.example.a61_viewpager2recyclerviewadapter_kotlin

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.a61_viewpager2recyclerviewadapter_kotlin.databinding.PagerItemBinding

class PagerBasicAdapter : RecyclerView.Adapter<PagerBasicViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PagerBasicViewHolder {
        return PagerBasicViewHolder(
            PagerItemBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            ))
    }

    override fun getItemCount(): Int = DataProvider.values.size

    override fun onBindViewHolder(holder: PagerBasicViewHolder, position: Int) {
        holder.bind(DataProvider.values[position])
    }
}